Support older versions of git

Phhere opened the pull request:
The Parameter -C is unknown in git 1.7.1 but --exec-path= can be used.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

jhseu commented:
Andrew, mind taking a look?

aselle commented:
Thank you for the patch.

You should squash your two commits into one using `git rebase -i` since the first commit is incorrect (it confused me at first.

Also, there is an internal committed change that will be sync'd soon that adds a 
`tensorflow/tools/git/gen_git_source.sh`. It would be great if you could make the change to that (which is used for the Makefile case to avoid the python dependency).

Phhere commented:
Ok, I will rebase it and wait for the internal change

Phhere committed:
Support older versions of git
The Parameter -C is unknown in git 1.7.1 but --git-dir= and --work-tree can be used.

Phhere commented:
I think the new version is ok

drpngx commented:
Jenkins, test this please.

drpngx commented:
@aselle , are we good to go?

drpngx commented:
(I don't think that the gen_git_source.sh in its present form requires modifications)
